signac/core/h5store.py

```diff
@@ -247,7 +247,11 @@ def __del__(self):
         self.close()
 
     def __enter__(self):
-        self.open()
+        try:
+            self.open()
+        except OSError as error:
+            if 'is already open' not in str(error):
```

Comment:
I'm not sure that I like tying this implementation to the specific error text. I would imagine that non-English versions of Python might not raise an error with this string. (Versuch es bitte?) I tried to find what raises this error (h5py? CPython? Something else?) but couldn't locate the source. Is there another way?

Reply:
Unfortunately the h5py package overall relies on generic RuntimeErrors for almost all error conditions. I don't know of a better way to limit the scope here. I don't thing that localized versions of the h5py package or the HDF5 library exist and therefore do not anticipate this to be a problem.